Sources/LanguageServerProtocol/Notifications/SourceKitFileStatusNotification.swift

```diff
@@ -0,0 +1,48 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Notification from the server about a per-file status update for the client.
+///
+/// - Parameter textDocument: the document the status notification is for
+/// - Parameter category: used to stylize the status in the editor (e.g. with an icon)
+/// - Parameter message: user visible description
+/// - Parameter clickTarget: URI to open on click if the editor supports it
+///
+/// Requires a private SourceKit extension.
+public struct SourceKitFileStatusNotification: TextDocumentNotification, Hashable {
+  public static let method: String = "textDocument/sourcekit.filestatus"
+
+  /// The `textDocument` this status notification is for.
+  public var textDocument: TextDocumentIdentifier
+
+  /// Status category, used to stylize the status in the editor (e.g. with an icon).
+  public var category: SourceKitFileStatusCategory
+
+  /// User visible description.
+  public var message: String
+
+  /// URI to open on click, if the editor supports it. Can be used to point
+  /// users towards an expanded description or error link.
+  public var clickTarget: DocumentURI?
```

Comment:
We should avoid terminology like "click".  The client defines the UI, not us, and we should provide information that is useful to a variety of clients.  How about "infoLink" or something similar?

Reply:
Done